Sets: Use of Reals set in place of S.Reals(11378)

shrutishrm512 opened the pull request:
Reals set is created in sets/**init**.py file which is equivalent to S.Reals by importing S from core module and deleting S so that file doesn't export S.

asmeurer commented on sympy/sets/__init__.py:
```diff
@@ -3,3 +3,6 @@
 from .conditionset import ConditionSet
+from ..core.singleton import S
+
+Reals = S.Reals
```
You need to add `del S` to the bottom so that this file doesn't export `S`.

shrutishrm512 committed:
Sets : Use of Reals set in place of S.Reals(11378)

Reals set is created in sets/__init__.py file which is equivalent to S.Reals by importing S from core module and deleting S so that file doesn't export S.

shrutishrm512 commented:
#11378

shrutishrm512 committed:
Sets: Use of Reals set in place of S.Reals(11378)

Reals set is created in sets/init.py file which is equivalent to S.Reals by importing S from core module and deleting S so that file doesn't export S.

shrutishrm512 commented:
#11686

shrutishrm512 commented:
@asmeurer Can you please explain why is getting failed even on including del S?

Shekharrajak commented on sympy/sets/__init__.py:
```diff
@@ -3,3 +3,7 @@
 from .contains import Contains
 from .conditionset import ConditionSet
+
+from ..core.singleton import S
```
Error is `ValueError: attempted relative import beyond top-level package`.
So I think this should work : 
`from sympy.core.singleton import S`

asmeurer replied:
I think it should stay as a relative import.

This error only occurs with Sphinx. Maybe there is a bug in Sphinx.

asmeurer replied:
You might try asking on StackOverflow about this. A quick Google search for the error message didn't turn anything up.

shrutishrm512 replied:
OK , i will search on google , but i didn't get any idea how all checks are getting failed?

jksuom replied:
[This question](http://stackoverflow.com/questions/72852/how-to-do-relative-imports-in-python) may be related.

asmeurer replied:
There's something specifically wrong with the way Sphinx is importing things. Or possibly we have some issue in our conf.py.

asmeurer replied:
I figured out the fix

```diff
diff --git a/doc/src/conf.py b/doc/src/conf.py
index 8ec9f7e..1e6503e 100644
--- a/doc/src/conf.py
+++ b/doc/src/conf.py
@@ -15,7 +15,7 @@
 import sympy

 # If your extensions are in another directory, add it here.
-sys.path = ['../sympy', 'ext'] + sys.path
+sys.path = ['ext'] + sys.path

 # General configuration
 # ---------------------
```

I don't know why `../sympy` was being added to the sys.path, but that's wrong. It was causing all the submodules of SymPy to be seen as top-level modules.  I guess Sphinx was importing `sets` (instead of `sympy.sets`) here, which explains the error message. `sympy` itself is added via `PYTHONPATH` being set in the Makefile.

moorepants commented:
@shrutishrm512 Can you address the issues here?

shrutishrm512 commented:
@moorepants I can't able to find out what are the problems here i was getting ! I added simply Reals which is equivalent to S.Reals after that i deleted S but i think on deletion of S, it is creating problem with other modules or packages

moorepants commented:
It sounds like the leading S is what Aaron desired in the related issue.

gxyd commented:
This is a PR for issue #11378.
